### PR TITLE
✨ PIC-1103 filter out messages for dummy courts

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**
Any messages indicating a court room > 50 (min-dummy-court-room property) will be ignored and not placed on the SQS queue

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt [hunk 1]
```diff
@@ -69,10 +70,26 @@ internal class ExternalDocRequestEndpointTest {
         verifyNoMoreInteractions(sqsService, telemetryService)
     }
 
+    @Test
+    fun `given a valid message with dummy court room then should enqueue the message and return the correct acknowledgement`() {
```
Comment (jhevans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt:74]: Should this not be
```suggestion
    fun `given a valid message with dummy court room then should *not* enqueue the message and return the correct acknowledgement`() {
```
?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt [hunk 2]
```diff
@@ -69,10 +70,26 @@ internal class ExternalDocRequestEndpointTest {
...
+        endpoint = buildEndpoint(setOf("B10JQ"), false, 5)
+        whenever(sqsService.enqueueMessage(contains("ExternalDocumentRequest")))
+            .thenReturn("a4e9ab53-f8aa-bf2c-7291-d0293a8b0d02")
+
+        val ack = endpoint.processRequest(externalDocument)
+
+        assertAck(ack)
+
+        verify(telemetryService).trackEvent(TelemetryEventType.COURT_LIST_MESSAGE_RECEIVED, customDimensionsMap)
+        verify(sqsService).enqueueMessage(anyString())
```
Comment (jhevans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt:85]: See previous comment, aren't we expecting these to be filtered out?

**Review by jhevans — COMMENTED** [review]
Looks good except one test which confused me - I'm probably being stupid but could you take a look and confirm 😄

File: src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt (hunk 2 above)
Comment (chris-faulkner-mojdt) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt:85]: I remove that and the test still passes. 😳 Yes - you are right.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt (hunk 1 above)
Comment (chris-faulkner-mojdt) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/endpoint/ExternalDocRequestEndpointTest.kt:74]: Correct - sorry

**Review by jhevans — APPROVED** [review]
LGTM 👍

**Pull request merged**